tests/topotests/ospf6-topo1/test_ospf6_topo1.py

```diff
@@ -242,73 +253,73 @@ def test_ospf6_converged():
         print("\nwaiting 15s for routes to populate")
         sleep(15)
 
+    # For debugging, uncomment the next line
+    # tgen.mininet_cli()
+
     # Make sure that all daemons are still running
-    for i in range(1, 5):
-        fatal_error = net['r%s' % i].checkRouterRunning()
-        assert fatal_error == "", fatal_error
+    if tgen.routers_have_failure():
+        assert tgen.errors == "", tgen.errors
+
+def compare_show_ipv6(rname, expected):
+    """
+    Calls 'show ipv6 route' for router `rname` and compare the obtained
+    result with the expected output.
+    """
+    tgen = get_topogen()
+    current = tgen.gears[rname].vtysh_cmd('show ipv6 route')
+
+    # Use just the 'O'spf lines of the output
+    linearr = []
+    for line in current.splitlines():
+        if re.match('^O', line):
+            linearr.append(line)
+
+    current = '\n'.join(linearr)
+
+    # Remove the link addresses
+    current = re.sub(r'fe80::[^ ]+', 'fe80::xxxx:xxxx:xxxx:xxxx', current)
```

Comment:
This IPv6 link-local address filtering seems like a recurring pattern. It would be nice to have a function in `lib/topotest.py` to do this everytime.

Example: https://github.com/FRRouting/frr/blob/master/tests/topotests/lib/topotest.py#L413

```py
def ip4_route_zebra(node, vrf_name=None):
    """
    Gets an output of 'show ip route' command. It can be used
    with comparing the output to a reference
    """
    if vrf_name == None:
        tmp = node.vtysh_cmd('show ip route')
    else:
        tmp = node.vtysh_cmd('show ip route vrf {0}'.format(vrf_name))
    output = re.sub(r" [0-2][0-9]:[0-5][0-9]:[0-5][0-9]", " XX:XX:XX", tmp)

    lines = output.splitlines()
    header_found = False
    while lines and (not lines[0].strip() or not header_found):
        if '> - selected route' in lines[0]:
            header_found = True
        lines = lines[1:]
```

Or we could use the `show ipv6 route json` command and simplify the test using `json_cmp`. Example here: https://github.com/FRRouting/frr/blob/master/tests/topotests/bfd-topo2/test_bfd_topo2.py#L144

```py
    # Check IPv6 routing tables.
    logger.info("Checking IPv6 routes for convergence")
    for router in tgen.routers().values():
        json_file = '{}/{}/ipv6_routes.json'.format(CWD, router.name)
        if not os.path.isfile(json_file):
            logger.info('skipping file {}'.format(json_file))
            continue

        expected = json.loads(open(json_file).read())
        test_func = partial(topotest.router_json_cmp,
                            router, 'show ipv6 route json', expected)
        _, result = topotest.run_and_expect(test_func, None, count=160,
                                            wait=0.5)
        assertmsg = '"{}" JSON output mismatches'.format(router.name)
        assert result is None, assertmsg```

Reply:
yes, good suggestion: I've added a lib function that canonicalizes that output, and use it in the test script now. I updated the reference/expected output files to match the lib function's output.